egret/model_library/unit_commitment/uptime_downtime.py

```diff
@@ -61,12 +62,24 @@ def enforce_fixed_commitments_rule(m,g,t):
 
 def _3bin_logic(model):
     
+    if is_var(model.UnitOn) and is_var(model.UnitStart) and is_var(model.UnitStop):
+        linear_expr = LinearExpression
+    else:
+        linear_expr = linear_summation
+
+    initial_time = value(model.InitialTime)
     def logical_rule(m,g,t):
-        if t==value(m.InitialTime):
-            return m.UnitOn[g, t] - m.UnitOnT0[g] == m.UnitStart[g,t] - m.UnitStop[g,t]
-        return m.UnitOn[g,t] - m.UnitOn[g,t-1] == m.UnitStart[g,t] - m.UnitStop[g,t]
+        if t==initial_time:
```

Comment:
Right -- I think to do this here (there's a mix of Params and Vars), we would need to have the time index of UnitOn to extend one back from model.TimePeriods. A similar thing could be done for model.PowerGenerated and model.MinimumPower, which would allow us to write the ramping constraints without and if/then.

One could also set up a unit commitment model without initial conditions, and let them be chosen by the solver. This may be the best way to kick off a Prescient run. That said, this change should be another PR.